Added /blockpvp /blockredstone and /blockedit

Commodore64x opened the pull request:
Issues referenced: #2017, #2016 and #2015
Note: This is live on the server, but this version has formatting, logic and grammar improvements

Commit:
Created /blockedit

Commit:
Created /blockpvp

Commit:
Created /blockredstone

Commit:
Data for /blockpvp and /blockedit

Commit:
Config Entry for /blockredstone

Commit:
New config entry for redstone use

Commit:
Events for redstone use

mayokek reviewed (approved):
Looks good. All it needs is testing.

Commit:
Grammar is important!

Commit:
Added handling for blockedit

Commit:
Added hanling for blockpvp

Commit:
Added in new services

mayokek commented on src/main/java/me/totalfreedom/totalfreedommod/command/Command_blockedit.java:
```diff
@@ -24,7 +24,7 @@ public boolean run(CommandSender sender, Player playerSender, Command cmd, Strin
         if (args[0].equals("list"))
         {
-            msg("Blocked block edits players:");
+            msg("Block modification is disallowed for these players:");
```
Maybe add 'the following' and remove 'these'?

Commit:
Frontdoor IP update (#2064)

Wild1145 committed:
Increased text limit. Resolves #1941 (#2039)



* Updated the comment as I forgot that

* Corrected the max limited based on updated issue from 250 to 256

Commit:
Update /smite (#2009)

This update makes smite messages stand out more in chat and to the smitten player.

Wild1145 committed:
Corrected player message for /disguisetoggle. Resolves #1872 (#2042)

Wild1145 committed:
Added a catch for if a player is caged and kicked. Resolves #1789 (#2041)

LiamTownsley commented on src/main/java/me/totalfreedom/totalfreedommod/command/Command_blockpvp.java:
```diff
@@ -0,0 +1,131 @@
+package me.totalfreedom.totalfreedommod.command;
```
This is already a command implemented by @marcocorriero

mayokek replied:
This PR was created before it was even implemented.

LiamTownsley reviewed (changes requested):
Currently I do not think that /blockredstone or /blockpvp are needed.

/blockredstone : If redstone is being abused ban the abuser.

/blockpvp : @marcocorriero already implemented this.

savnithyt commented:
```
import org.bukkit.event.block.BlockPistonExtendEvent;
import org.bukkit.event.block.BlockPistonRetractEvent;
import org.bukkit.event.block.BlockRedstoneEvent;
```

Needs to be added in EventBlocker.java other than that the plugin works fine!

savnithyt commented:
```
import org.bukkit.event.block.BlockPistonExtendEvent;
import org.bukkit.event.block.BlockPistonRetractEvent;
import org.bukkit.event.block.BlockRedstoneEvent;
```
^ That needs to be added in EventBlocker.java, other than that the code works fine.

Wild1145 reviewed (changes requested):
This PR is currently not building for me. Can you look into this please?

Commit:
Fixed import errors

ghost commented:
@Wild1145 PR should build now.

Wild1145 commented:
@Commodore64x This is still failing the TravisCI Integration tests.

JeromSar commented:
@Commodore64x As with the other branch. Please merge with `devel` to fix the Travis CI issue.

Commit:
Wild's changes (#2)

* Moved all dependencies over to proper resoultion. Currently experiencing an issue with TF-WorldEdit not pulling custom additions

* Final fix to the dependency resolution. Reverted minor changes to WorldEdit handling

* Added Travis YAML File for CI#

* Fixed issues with POM file - Everything is resolving properly and CI is working

* Update pom.xml

Updated Libs Disguise version

* Update pom.xml

* Updated Libs Disguise Version Dependency

It will now always pull the latest 1.11 branch commit, this should remain build-able and stable as it is a TF Project.

* Update README.md

* Updated to 1.12 R01 for dependencies

* Updated dependencies removing Libs Disguise due to unstable upstream repos

* Resolved dependency issues - Building correctly

* Removed dependency to Craftbukkit and Spigot server code.

This breaks whitelist and onlinemode toggle.

* Add .travis.yml file for Travis CI integration

* Update Travis CI status image link.

* Logviewer registration overhaul.

* Set up Travis CI automatic build uploads to SFTP server.

ghost commented:
@JeromSar @Wild1145 Should work with travis now.
